routers/api/v1/repo/wiki.go

```diff
@@ -0,0 +1,501 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package repo
+
+import (
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/convert"
+	"code.gitea.io/gitea/modules/git"
+	api "code.gitea.io/gitea/modules/structs"
+	"code.gitea.io/gitea/modules/util"
+	"code.gitea.io/gitea/modules/web"
+	wiki_service "code.gitea.io/gitea/services/wiki"
+)
+
+// NewWiki response for wiki create request
+func NewWiki(ctx *context.APIContext) {
+	// swagger:operation POST /repos/{owner}/{repo}/wiki/new repository repoCreateWikiPage
+	// ---
+	// summary: Create a wiki page
+	// consumes:
+	// - application/json
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo
+	//   type: string
+	//   required: true
+	// - name: body
+	//   in: body
+	//   schema:
+	//     "$ref": "#/definitions/CreateWikiPageOptions"
+	// responses:
+	//   "204":
+	//     "$ref": "#/responses/empty"
+	//   "400":
+	//     "$ref": "#/responses/error"
+	//   "403":
+	//     "$ref": "#/responses/forbidden"
+
+	form := web.GetForm(ctx).(*api.CreateWikiPageOptions)
+
+	if util.IsEmptyString(form.Title) {
+		ctx.Error(http.StatusBadRequest, "emptyTitle", nil)
+		return
+	}
+
+	wikiName := wiki_service.NormalizeWikiName(form.Title)
+
+	if len(form.Message) == 0 {
+		form.Message = ctx.Tr("repo.editor.add", form.Title)
```

Comment:
Hmm.
I am not a particular fan of displaying (localized) content here.
Most wikis are written in English only, so the current user locale should not have an effect on the outcome.
That is exactly the same as #14845 for wiki pages.
As I have just noticed, Gitea does not allow empty commit messages, so we have to insert some content.
But I am against a localized string, instead I would simply use
```suggestion
		form.Message = form.Title
```

Reply:
It's just copied from the web router - it also localizes every commit

Edit:
I definitely can understand you, but it's a bit strange if you have a commit message `Home`... Maybe you could just go with the English version `Add/edit Home`

Reply:
> It's just copied from the web router - it also localizes every commit

That is another problem.
I agree with the commit message.
I thought `form.Message` referred to the page content. Go ahead with the English version.

Reply:
Fixed https://github.com/go-gitea/gitea/pull/17278/commits/8d568a448e1d11d8373650d2442f28f776fa17da